htmlmaker_preprocessing.ps1

```diff
@@ -32,35 +35,45 @@ Function LogWrite
    Add-content $Logfile -value "$logstring"
 }
 
-# if original file name is a .doc file
-$fileType = ".doc"
-If ($filenameSplit -eq $filename + $fileType)
-{
-	# Converts a word .doc in bookmaker_tmp to a .docx
-	# so you have to have already run tmparchive.rb
-	write-host "Converting $filenameSplit to .docx from $fileType"
 
-	$SaveFormat = "microsoft.office.interop.word.WdSaveFormat" -as [type]
-	$word = New-Object -ComObject word.application
-	$word.visible = $false
+# Converts a word .doc in bookmaker_tmp to a .docx
+# so you have to have already run tmparchive.rb
+write-host "Converting $filenameSplit to .docx from $fileType"
+
+$SaveFormat = "microsoft.office.interop.word.WdSaveFormat" -as [type]
+$word = New-Object -ComObject word.application
+$word.visible = $false
+
+# Make sure "Store random number..." setting is off 
+$word.Options.StoreRSIDOnSave = $false
+
+$doc = $word.documents.open($docpath + $fileType)
 
-	$doc = $word.documents.open($docpath + $fileType)
-	$wdFormatDocx = 16  # wdFormatDocumentDefault is docx, reference number is 16
+# turn off track changes, accept all changes and delete all comments
+# have to do here, or we don't solve dumb XML revision problem
+$doc.TrackRevisions = $false
+$doc.Revisions.AcceptAll
+Foreach($comment in $doc.Comments)
+{
+    $comment.Delete
+}
+
+# save file as .docx
+$wdFormatDocx = 16  # wdFormatDocumentDefault is docx, reference number is 16
 	
-	# Have to add [ref]s for certain versions of powershell (2.0), 
-    # we'll see which way works on server
-	# https://richardspowershellblog.wordpress.com/2012/10/15/powershell-3-and-word/
-	# so only use one or the other of these next two lines
-	# $doc.saveas($docpath, $wdFormatDocx)
-	$doc.saveas([ref]$docpath, [ref]$wdFormatDocx)
+# Have to add [ref]s for certain versions of powershell (2.0), 
+# we'll see which way works on server
+# https://richardspowershellblog.wordpress.com/2012/10/15/powershell-3-and-word/
+# so only use one or the other of these next two lines
+# $doc.saveas($docpath, $wdFormatDocx)
+$doc.saveas([ref]$docpath, [ref]$wdFormatDocx)
```

Comment:
.SaveAs should overwrite by default. I'm not 100% sure that the save will happen if nothing has been changed in the original file, though. If that's the case, see if you can set the `.Saved` property to `$false` first.